src/hotspot/share/gc/shenandoah/shenandoahVerifier.cpp

```diff
@@ -800,7 +800,7 @@ void ShenandoahVerifier::verify_at_safepoint(const char* label,
         break;
       case _verify_gcstate_evacuation:
         enabled = true;
-        expected = ShenandoahHeap::HAS_FORWARDED | ShenandoahHeap::EVACUATION;
+        expected = ShenandoahHeap::EVACUATION;
```

Comment:
May be the semantics of these states should be explained via a state diagram where these are defined using "may" and "must" modalities, so the semantics are clear and there isn't ambiguity. ("has" to me feels like a "must" modality, in the modal logic sense; `EVACUATION` could similarly be clarified with a `MAY` or `MUST` modal sense.)

I am rambling, but hopefully you get the idea of what I am trying to articulate :-)

Reply:
I think the `expected` state here is effectively a `MUST`.